src/stash/src/postgres.rs

```diff
@@ -1332,6 +1332,17 @@ impl DebugStashFactory {
                 tracing::error!("postgres stash connection error: {e}");
             }
         });
+        // Running tests in parallel has been causing some deadlock/starvation issue that we haven't
+        // been able to figure out. For some reason, uncommitted transactions are being left open
+        // which causes the schema cleanup in the Drop impl to block for an enormous amount of time
+        // (many minutes). Setting a small idle transaction timeout globally seems to resolve the
+        // issue somehow. This is a gross hack and we don't really understand what's going on, but
+        // for now it resolves issues in CI while we debug further.
+        client
+            .batch_execute(
+                "SET CLUSTER SETTING sql.defaults.idle_in_transaction_session_timeout TO '3s'",
```

Comment:
3s seems like it'd flake because of being too low. But if you run CI multiple times and it's fine then lgtm. Hopefully we can fix the underlying problem and remove this.

Reply:
I just pushed 1s, we'll see how it does.

Note it just flaked but the error wasn't a timeout:

```
thread 'test_auth_expiry' panicked at 'called `Result::unwrap()` on an `Err` value: stash error: postgres: db error: ERROR: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE - failed preemptive refresh due to a conflict: intent on key /NamespaceTable/30/1/100/113/"collections"/4/1): "sql txn" meta={id=ac0e4628 key=/Table/3/1/259/2/1 pri=0.01736586 epo=0 ts=1695853667.841923687,2 min=1695853667.298543540,0 seq=20} lock=true stat=PENDING rts=1695853667.298543540,0 wto=false gul=1695853667.798543540,0 
```